core/src/main/scala/tech/beshu/ror/utils/SSLCertParser.scala

```diff
@@ -51,6 +52,21 @@ object SSLCertParser extends Logging {
         _ => true
       )
 
+  def customTrustManagerFrom(config: SslConfiguration): Option[TrustManagerFactory] = {
+    config.truststoreFile
+      .map { file =>
+        logger.info(s"Using custom truststore: '${file.getName}'")
+        val truststore = KeyStore.getInstance(KeyStore.getDefaultType)
+        truststore.load(
+          new FileInputStream(file),
```

Comment:
Inspired by below `tryRun` function. We open `FileInputStream` but never close. Similar in all `SSLNetty4HttpServerTransport` classes for `ByteArrayInputStream`. Is there some specific reason we do not close streams?

Reply:
no, IMO we should close the stream. Maybe you can use cat's effect Resource to handle it in a neat way